documentation/ADRs/025_log _level_standards.md

```diff
@@ -0,0 +1,77 @@
+## Title
+*Log Level Standards*
+
+| ADR Info            | Details           |
+|---------------------|-------------------|
+| Subject             | Logging Levels Configuration  |
+| ADR Number          | 025   |
+| Status              | Proposed  |
+| Author              | Simon   |
+| Date                | 30.10.2024     |
+
+## Context
+We aim to establish a new benchmark in MLOps for early warning systems (EWS), emphasizing robust and transparent logging practices. The conflict forecasting pipeline must have a comprehensive logging strategy to support the continuous quality assurance, real-time monitoring, and rapid model updates critical for high-stakes decision-making and early action. This ADR specifically addresses the standardized use of log levels within the pipeline, which helps the team capture relevant system states and provides clear visibility into operations, potential issues, and crucial decision points across the pipeline.
+
+The following log levels—DEBUG, INFO, WARNING, ERROR, and CRITICAL—are configured to ensure appropriate information is logged for various scenarios, supporting both ongoing development and long-term system monitoring and troubleshooting.
+
+## Decision
+The following log levels are implemented as standard for the conflict forecasting pipeline:
+
+### Overview
+The system’s log levels are structured as follows:
+
+1. **DEBUG**: Provides detailed diagnostic information, primarily used during development and debugging.
+2. **INFO**: Captures general system information about normal operations to give an overview without verbosity.
+3. **WARNING**: Logs potentially problematic situations that require attention but do not yet impact execution.
+4. **ERROR**: Indicates issues where specific processes fail but the overall system remains operational.
+5. **CRITICAL**: Records severe errors that require immediate attention as they could lead to system failures or data loss.
+
+### Examples and Use Cases
+
+- **DEBUG**:
+  - Example: Logging input data shapes, intermediate transformations, or model hyperparameters during experimentation phases.
+  - Use: Essential for development and debugging stages, enabling the team to trace and diagnose precise pipeline states.
```

Comment:
Example: get_data.py provides the shape of data if exists when loading the data.
https://github.com/prio-data/views_pipeline/blob/8bedc53fc9fdf1070e894777a4f113438008439c/models/orange_pasta/src/dataloaders/get_data.py#L13

Reply:
adeded